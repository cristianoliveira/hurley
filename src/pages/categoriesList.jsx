import React, { Component } from 'react';
import { Link } from 'react-router';

class CategoriesList extends Component {
  render() {
    return (
      <div>
        <h1> Lista de Categorias </h1>
      </div>
    );
  }
}

export default CategoriesList;
