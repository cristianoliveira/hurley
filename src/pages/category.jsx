import React, { Component } from 'react';
import { Link } from 'react-router';
import Helmet from 'react-helmet';
import ChannelCard from './../components/ChannelCard'
import FeedcastLoader from './../components/FeedcastLoader'
import feedcastApi from 'feedcast-client';
import helpers from './../scripts/helpers'

class Category extends Component {

  constructor(props) {
    super(props);

    let { lc } = helpers.localize(this)

    this.state = {
      channels : [],
      icon: null,
      slug: null,
      title: null,
      populated: false,
      lc
    }
  }


  componentWillUnmount(){
    this._isMounted = false
  }

  componentDidMount() {
    this._isMounted = true
    this.updatePage(this.props)
  }


  updatePage(props){
    const { page } = props.params
    const slug = page && page.length > 0 ? page : ''
    this.setState({ populated: false }, ()=>{
      feedcastApi
        .getEpisodesByCategory({slug})
        .then( data => {
          this.setState({
            populated: true,
            ...data
          });
        })
    });
  }



  listChannels(){
    let { channels } = this.state;
    return channels
            .filter(c => c.listed)
            .map( (c, n) => (<ChannelCard key={n} data={c}/>));
  }


  componentWillUpdate(nextProps) {
    if( this.props.params.page != nextProps.params.page)
      this.updatePage(nextProps);
  }



  render() {
    const channelList = this.listChannels()

    return this.state.populated ? (
      <div>
        <Helmet
          title={`Feedcast`}
          meta={[
            {property: 'og:title',
            content: `Feedcast ${ this.state.populated ? '| ' + this.state.title : '' }`},
          ]} />
        <h1>{helpers.translate(this.state.title)} </h1>
        <div className="feedcast__channel-list feedcast__channel-list--byCategory">
          {channelList}
        </div>
      </div>
    ) : (<FeedcastLoader/>);
  }
}

export default Category;
