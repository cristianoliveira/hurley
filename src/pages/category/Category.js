import React, { Component } from 'react';
import PropTypes from 'prop-types';

import Helmet from 'react-helmet';
import ChannelCard from 'app/components/ChannelCard'

class Category extends Component {
  static propTypes = {
    category: PropTypes.object.isRequired,
  }

  static defaultProps = {
    category: { channels: [] },
  }

  listChannels(){
    if(!this.props.category.channels) { return null; }

    return this.props.category.channels
            .filter(c => c.listed)
            .map((channel, i) => (<ChannelCard key={i} data={channel}/>));
  }

  render() {
    const channelList = this.listChannels()

    return (
      <div>
        <Helmet
          title={`Feedcast`}
          meta={[
            {property: 'og:title',
            content: `Feedcast ${ this.props.category.title }`},
          ]} />
        { this.props.title ? (<h1>this.props.title</h1>) : null }
        <div className="feedcast__channel-list feedcast__channel-list--byCategory">
          {channelList}
        </div>
      </div>
    );
  }
}

export default Category;
