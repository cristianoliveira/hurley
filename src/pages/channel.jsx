import React, { Component } from 'react';
import { Link } from 'react-router';
import Helmet from 'react-helmet';

import feedcastApi from './../scripts/feedcastApi'
import helpers from './../scripts/helpers'

import './../styles/channel.sass'
import ChannelEpisodes from './../components/ChannelEpisodes.jsx'
import FeedcastLoader from './../components/FeedcastLoader.jsx'


class Channel extends Component {
  constructor(){
    super();

    this.state = {
      categories : [],
      description: '',
      feed_url: '',
      image_url: '',
      listed: false,
      slug: '',
      synchronization_status: '',
      synchronization_status_message: '',
      title: '',
      uuid: '',
      populated: false
    }
  }

  componentDidMount() {
    this._isMounted = true
  }

  componentWillMount() {
    this._isMounted = false
    const { uuid } = this.props.params
    feedcastApi
      .getChannelInfo({ uuid })
      .then(data => {
        data.populated = true
        this.setState(data)
      })
  }



  render() {
    let { categories, description,
      image_url, title, uuid } = this.state

    let { page } = this.props.params

    categories = categories.map( (i,n) => (
      <Link key={n} to={`/category/${i.slug}`}>
        <i className={`fa fa-${i.icon}`}></i> {helpers.translate(i.title)}
      </Link>
    ));

    let metaTitle = this.state.populated ? `| ${this.state.title}` : ``

    return this.state.populated ? (
      <div className="feedcast__channelPage">
        <Helmet
          title={`Feedcast ${metaTitle}`}
          meta={[
            {property: 'og:title',
            content: `Feedcast ${metaTitle}`},
          ]} />
        <div className="feedcast__channelInfo">
          <div className="feedcast__channelInfo-header">
            <img className="feedcast__channelInfo-img" src={image_url}/>
            <h1 className="feedcast__channelInfo-title">{title}</h1>
          </div>
          <div className="feedcast__channelInfo-body">
            <p className="feedcast__channelInfo-description">{description}</p>
            <div className="feedcast__channelInfo-categories">{categories}</div>
          </div>
        </div>
        <ChannelEpisodes data={{uuid, page}}/>
      </div>
    ) : (
      <FeedcastLoader />
    );
  }
}

export default Channel;
