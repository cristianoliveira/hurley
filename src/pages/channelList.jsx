import React, { Component } from 'react';
import { Link } from 'react-router';
import './../styles/channelList.css';

import feedcastApi from './../scripts/feedcastApi'

class ChannelList extends Component {

  constructor(){
    super();

    this.state = {
      total:null,
      page: 1,
      per_page: 24,
      channels: []
    }

  }


  updateList(){
    let {page, per_page} = this.state

    feedcastApi
      .getChannels({page, per_page})
      .then(data => {
      this.setState({
        page: parseInt(page),
        channels: data.channels,
        total: data.total
      });
    })
  }


  componentWillUpdate(nextProps) {
    if( this.props.params.page != nextProps.params.page)
      this.updatePage(nextProps);
  }


  componentWillMount(){
    this.updatePage(this.props)
  }


  updatePage(props){
    const { params } = props
    switch(!0){
      case params.page !== undefined &&
        parseInt(params.page) !== this.state.page &&
        parseInt(params.page) >= 0:

        this.setState({ page: parseInt(params.page) }, () => {
          this.updateList()
        });

      break;

      case Object.keys(params).length === 0:
        this.setState({ page: 1 }, () => {
          this.updateList()
        });
      break;

      case params.page !== undefined &&
        parseInt(params.page) == 1 &&
        parseInt(params.page) == this.state.page &&
        this.state.channels.length === 0:

        this.updateList();

      break;
    }
  }


  navigation(){
    let { channels, total, per_page, page} = this.state;
    let totalPages = parseInt(total / per_page);

    if(total % per_page > 0)
      totalPages++;

    let pages = []


    pages[1] = 1
    pages[page - 1] = page - 1
    pages[page] = page
    pages[page + 1] = page + 1
    pages[totalPages] = totalPages

    let buttons = pages
                    .filter(i => i > 0 && i <= totalPages)
                    .map(i=>(
        <Link
          key={i}
          to={`channels/${i}`}
          className={i == page ? 'active':''}>
          <button>{i}</button>
        </Link>
      ))

    return totalPages > 0 ? (
      <div className="feedcast__channel-navigation">
        <Link to={`channels/${
          this.state.page > 1?
          (this.state.page - 1) : 1}`} >
          <button>
            <i className="fa fa-angle-double-left"></i>
          </button>
        </Link>
        {buttons}
        <Link to={`channels/${
          this.state.page < totalPages?
          (this.state.page + 1) : totalPages}`}>
          <button>
            <i className="fa fa-angle-double-right"></i>
          </button>
        </Link>
      </div>
    ) : ''
  }


  listChannels(){
    let { channels, total, per_page} = this.state;
    return channels.map( c => (
      <Link key={c.uuid} to={`channel/${c.uuid}`}>
        <div className="feedcast__channel" id={c.uuid}>
          <div className="feedcast__channel-title">
            <h3>{c.title}</h3>
          </div>
          <img src={c.image_url} className="feedcast__channel-img" />
          <div className="feedcast__channel-overlay"></div>
        </div>
      </Link> ));
  }




  render() {
    let channelList = this.listChannels()
    let Navigation = this.navigation();
    return (
      <div className="feedcast__channel-list">
        {channelList}
        {Navigation}
      </div>
    );
  }
}

export default ChannelList;
