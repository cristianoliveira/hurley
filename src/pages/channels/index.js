import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';

import { requestChannels } from 'app/actions/channels';
import FeedcastLoader from 'app/components/FeedcastLoader';
import ChannelList from './ChannelList';

class ChannelListContainer extends Component {
  componentDidMount() {
    this.request(this.props.page);
  }

  request(page) {
    this.props.dispatch(requestChannels({
      page: page,
      perPage: this.props.perPage,
    }));
  }

  render() {
    const page = parseInt(this.props.params.page || "1");
    if (page !== this.props.page) {
      this.request(page);
    }

    if (this.props.isFetching) {
      return <FeedcastLoader />;
    }

    return <ChannelList {...this.props} />;
  }
}

export { ChannelList };

const mapStateToProps = (state) => state.channels;
export default connect(mapStateToProps)(ChannelListContainer);
