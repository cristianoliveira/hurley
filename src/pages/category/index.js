import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';

import { requestCategoryInfo } from 'app/actions/categories';
import FeedcastLoader from 'app/components/FeedcastLoader';
import Category from './Category';

class CategoryContainer extends PureComponent {
  static defaultProps = {
    categories: [],
    isFetching: true,
    params: {},
  }

  fetchData(slug) {
    this.props.dispatch(requestCategoryInfo(slug));
  }

  render() {
    if (this.props.isFetching) {
      return <FeedcastLoader />;
    }

    const category = this.props.categories[0] || {};
    if (category.slug !== this.props.match.params.slug) {
      this.fetchData(this.props.match.params.slug);
    }

    return <Category {...this.props} category={this.props.categories[0]} />;
  }
}

const mapStateToProps = (state) => state.categories;

export { Category };
export default connect(mapStateToProps)(CategoryContainer);
