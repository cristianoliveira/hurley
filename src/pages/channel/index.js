import React, { PureComponent } from 'react';
import { connect } from 'react-redux';

import { requestChannelInfo } from 'app/actions/channels';
import FeedcastLoader from 'app/components/FeedcastLoader';
import Channel from './Channel';

class ChannelContainer extends PureComponent {

  static defaultProps = {
    episodes: [],
    isFetching: true,
    page: 1,
    params: {},
  }

  componentDidMount() {
    const uuid = this.props.params.uuid;
    if (uuid !== this.props.uuid) {
      this.fetchData(uuid, this.props.page);
    }
  }

  fetchData(uuid, page) {
    this.props.dispatch(requestChannelInfo(uuid, page));
  }

  render() {
    const page = parseInt(this.props.params.page || '1');
    if (page !== this.props.page) {
      this.fetchData(
        this.props.params.uuid,
        page,
      );
    }

    if (this.props.isFetching) {
      return <FeedcastLoader />;
    }

    return <Channel {...this.props} channel={this.props.channels[0]} />;
  }
}

const mapStateToProps = (state) => state.channels;

export { Channel };
export default connect(mapStateToProps)(ChannelContainer);
