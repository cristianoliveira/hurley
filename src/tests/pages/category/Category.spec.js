import React from 'react';
import renderer from 'react-test-renderer';
import { withStore } from 'app/tests/helpers';

import { Category } from 'app/pages/category';

describe('Category', () => {
  describe('when has no data', () => {
    it("renders the no episode", () => {
      const component = renderer.create(<Category />);
      expect(component).toMatchSnapshot();
    })
  });

  describe('when has a list of category', () => {
    const state = {
      category: {
          uuid: "asdasdadasda",
          slug: "pensando-rpg",
          title: "Pensando RPG",
          description: "",
          feed_url: "http://feeds.soundcloud.com/users/soundcloud:users:257525576/sounds.rss",
          image_url: "https://img.feedcast.io/http%3A%2F%2Fi.imgur.com%2FmYRfJk7.png/300/300",
          listed: true,
          synchronization_status: "success",
          synchronization_status_message: "",
          channels: [],
        }
      ,
      page: 1,
      perPage: 130,
      total: 1000,
    }

    it('render the category', () => {
      const component = renderer.create(withStore(<Category {...state} />));
      expect(component).toMatchSnapshot();
    });
  });
})
