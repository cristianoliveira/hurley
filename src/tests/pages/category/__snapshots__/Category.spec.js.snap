// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Category when has a list of category render the category 1`] = `
<div>
  <h1>
     
  </h1>
  <div
    className="feedcast__channel-list feedcast__channel-list--byCategory"
  />
</div>
`;

exports[`Category when has no data renders the no episode 1`] = `
<div>
  <h1>
     
  </h1>
  <div
    className="feedcast__channel-list feedcast__channel-list--byCategory"
  />
</div>
`;
